PMSCore/UserBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿// UserBase.cs
+namespace PMSCore
+{
+    /// <summary>
+    /// Represents the base class for all user types.
+    /// </summary>
+    public abstract class UserBase : IUser
```

Comment:
IUser need to have the following methods
CreateEntity()
ChangeEntityProperty()
ChangeEntityStatus()
DeleteEntity()
AssociateWithEntity()
DisassociateWithEntity()
AssociateEntityToEntity()
DissociateEntityFromEntity()

Then UserBase can implement the most generic ones
Every concrete user can override the methods if necessary